plan/physical_plan_builder.go

```diff
@@ -43,7 +44,78 @@ const (
 // JoinConcurrency means the number of goroutines that participate in joining.
 var JoinConcurrency = 5
 
-func getRowCountByIndexRanges(sc *variable.StatementContext, table *statistics.Table, indexRanges []*IndexRange, indexInfo *model.IndexInfo) (uint64, error) {
+func getRowCountByIndexRanges(sc *variable.StatementContext, table *statistics.Table, indexRanges []*IndexRange, indexInfo *model.IndexInfo, tableInfo *model.TableInfo) (uint64, error) {
+	var offset int
+	for i := range tableInfo.Indices {
+		if tableInfo.Indices[i].Name.L == indexInfo.Name.L {
+			offset = i
+			break
+		}
+	}
+	if len(table.Indices[offset].Numbers) == 0 {
+		return getRowCountByPseudoIndexRanges(sc, table, indexRanges, indexInfo)
```

Comment:
The `indexRanges` is not pseudo, the stats is pseudo.
`getPseudoRowCountByIndexRanges` is better.